src/plugins/index_pattern_field_editor/public/components/preview/field_list/field_list.tsx

```diff
@@ -21,22 +25,36 @@ const ITEM_HEIGHT = 40;
 const SHOW_MORE_HEIGHT = 40;
 const INITIAL_MAX_NUMBER_OF_FIELDS = 7;
 
-export interface Field {
-  key: string;
-  value: string;
-}
+export type DocumentField = FieldPreview & {
+  isPinned?: boolean;
+};
 
 interface Props {
   height: number;
+  searchValue?: string;
+}
+
+/**
+ * Copied from https://stackoverflow.com/a/9310752
+ */
+function escapeRegExp(text: string) {
+  return text.replace(/[-\[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
```

Comment:
What does this do? Would be best to document directly in the code rather than linking.

Reply:
Added a comment 👍